pkg/kubelet/lifecycle/predicate.go

```diff
@@ -221,20 +226,25 @@ func (e *PredicateFailureError) GetReason() string {
 }
 
 // GeneralPredicates checks a group of predicates that the kubelet cares about.
-func GeneralPredicates(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo) ([]PredicateFailureReason, error) {
+func GeneralPredicates(pod *v1.Pod, nodeInfo *schedulerframework.NodeInfo, nodeCapacity v1.ResourceList) ([]PredicateFailureReason, error) {
 	if nodeInfo.Node() == nil {
 		return nil, fmt.Errorf("node not found")
 	}
 
-	var reasons []PredicateFailureReason
-	for _, r := range noderesources.Fits(pod, nodeInfo) {
-		reasons = append(reasons, &InsufficientResourceError{
-			ResourceName: r.ResourceName,
-			Requested:    r.Requested,
-			Used:         r.Used,
-			Capacity:     r.Capacity,
-		})
-	}
+	// retry if ephemeral storage doesn't fit for not initialized
+	wait.PollImmediate(1*time.Second, 2*time.Minute, func() (bool, error) {
```

Comment:
I'm not sure what the right timeout is here. Is 2 minutes too low?

This is definitely the right approach here!

Reply:
I'm not sure how long the waiting should be.
Are 5 minutes enough?

Reply:
I'll test this this week and report back!